spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -265,24 +265,51 @@ public String getLog(String platformName, String taskId) {
 
 	@Override
 	public void stopTaskExecution(Set<Long> ids) {
-		final Map<String, Object> auditData = new LinkedHashMap<>();
 		logger.info("Stopping {} task executions.", ids.size());
-		Set<TaskExecution> taskExecutions = getTaskExecutions(ids);
-		validateExternalExecutionIds(taskExecutions);
-		Set<Long> childTaskExecutionIds = this.dataflowTaskExecutionDao.findChildTaskExecutionIds(ids);
-		Set<TaskExecution> childTaskExecutions = getTaskExecutions(childTaskExecutionIds);
-		validateExternalExecutionIds(childTaskExecutions);
+
+		Set<TaskExecution> taskExecutions = getValidStopExecutions(ids);
+		Set<TaskExecution> childTaskExecutions = getValidStopChildExecutions(ids);
 		for (TaskExecution taskExecution : taskExecutions) {
 			cancelTaskExecution(taskExecution);
 		}
-
 		childTaskExecutions.forEach(childTaskExecution -> {
 			cancelTaskExecution(childTaskExecution);
 		});
 
-		long numberOfExecutionsStopped = ids.size() + childTaskExecutionIds.size();
+		updateAuditInfoForTaskStops(taskExecutions.size() + childTaskExecutions.size());
+	}
+
+	@Override
+	public void stopTaskExecution(Set<Long> ids, String platform) {
```

Comment:
There is code duplication between this method and stopTaskExecution(ids), can be removed.